packages/dd-trace/src/opentracing/tracer.js

```diff
@@ -37,9 +38,14 @@ class DatadogTracer extends Tracer {
     this._logInjection = config.logInjection
     this._analytics = config.analytics
     this._prioritySampler = new PrioritySampler(config.env)
-    this._writer = new Writer(this._prioritySampler, config.url)
-    this._recorder = new Recorder(this._writer, config.flushInterval)
-    this._recorder.init()
+
+    if (config.experimental !== undefined && config.experimental.exporter === constants.LOG_EXPORTER) {
```

Comment:
Unnecessary check for `undefined`.

Reply:
If I remove the check, it breaks a lot of tests in a bunch of different files which have been initializing tracer with an incomplete config.

Reply:
I would say to fix the tests as they should not force workarounds in code.

Reply:
I just double checked, I was wrong about that test, just going to follow your original suggestion.